src/core/weightedGraph2.js

```diff
@@ -37,89 +65,252 @@ export type WeightedEdge = {|
  * WeightedGraph2 should completely replace it.
  */
 export class WeightedGraph2 {
-  _nodeWeights: NodeTrie<NodeWeight>;
-  _edgeWeights: EdgeTrie<EdgeWeight>;
-  graph: Graph;
+  _weights: WeightsI;
+  _graph: Graph;
+
+  constructor(graph?: Graph, weights?: WeightsI) {
+    this._graph = graph ? graph : new Graph();
+    this._weights = weights ? weights : Weights();
+  }
 
   /**
    * Set the weight for a node address prefix.
    */
-  setNodePrefixWeight(
-    _unused_prefix: NodeAddressT,
-    _unused_w: NodeWeight
-  ): this {
-    throw new Error("method not implemented");
+  setNodePrefixWeight(prefix: NodeAddressT, w: NodeWeight): this {
```

Comment:
`setNodePrefixWeight` _is_ `setNodeWeight` on `WeightsI`.  It seems redundant to rewrite the function here, when we can simply set it:
```js
// Even with classes, you can just set this in the constructor
constructor(graph?: Graph, weights?: WeightsI) {
  ...
  setNodePrefixWeight = weights.setNodeWeight;
  setEdgePrefixWeight = weights.setEdgeWeight;
}
```